plugins/module_utils/omevv_utils/omevv_firmware_utils.py

```diff
@@ -254,3 +274,337 @@ def delete_firmware_repository_profile(self, profile_id):
+
+        Raises:
+            None
+        """
+        response = self.omevv.invoke_request('GET', BASELINE_PROFILE_URI.format(vcenter_uuid=vcenter_uuid))
+        if response.success:
+            return response.json_data
+        else:
+            return []
+
+    def get_baseline_profile_by_id(self, profile_id, vcenter_uuid):
+        """
+        Retrieves all baseline profile Information.
+
+        """
+        resp = self.omevv.invoke_request(
+            "GET", BASELINE_PROFILE_URI.format(vcenter_uuid=vcenter_uuid) + "/" + str(profile_id))
+        return resp
+
+    def get_baseline_profile_by_name(self, profile_name, vcenter_uuid):
+        """
+        Retrieves all baseline profile Information.
+
+        """
+        profiles = self.get_baseline_profiles(vcenter_uuid)
+        profile_exists = self.search_baseline_profile_name(profiles, profile_name)
+
+        if profile_exists:
+            return profile_exists
+
+        else:
+            return {}
+
+    def create_job_schedule(self, days, time):
+        """
+        Creates a job schedule based on provided days and time.
+
+        Args:
+            days (list): List of days selected for the job schedule.
+            time (str): The time to be set for the job schedule.
+
+        Returns:
+            dict: A dictionary representing the job schedule.
+        """
+        if days and time:
+            days_selected = set(days)
+
+            if "all" in days_selected:
+                return {
+                    "monday": True,
+                    "tuesday": True,
+                    "wednesday": True,
+                    "thursday": True,
+                    "friday": True,
+                    "saturday": True,
+                    "sunday": True,
+                    "time": time
+                }
+            else:
+                return {
+                    "monday": "monday" in days_selected,
+                    "tuesday": "tuesday" in days_selected,
+                    "wednesday": "wednesday" in days_selected,
+                    "thursday": "thursday" in days_selected,
+                    "friday": "friday" in days_selected,
+                    "saturday": "saturday" in days_selected,
+                    "sunday": "sunday" in days_selected,
+                    "time": time
+                }
+        return None
+
+    def get_current_job_schedule(self, profile_id, vcenter_uuid):
+        """
+        Retrieves the current job schedule for a baseline profile.
+
+        Args:
+            profile_id (str): The ID of the profile for which to retrieve the schedule.
+            vcenter_uuid (str): The UUID of the vCenter.
+
+        Returns:
+            dict: Job schedule details if available.
+        """
+        resp = self.omevv.invoke_request(
+            "GET", DRIFT_URI.format(vcenter_uuid=vcenter_uuid) + "/" + str(profile_id)
+        )
+        return resp
+
+    def search_baseline_profile_name(self, data, profile_name):
```

Comment:
perform same in function, avoid creating new function for this.

Reply:
Done